src/test/simple_spin.cc

```diff
@@ -46,7 +46,8 @@ TEST(SimpleSpin, Test1)
   // Should also work with pass-by-reference:
   // (Note that we don't care about cross-threading here as-such.)
   counter = 0;
-  ASSERT_EQ(counter, 0);
+  uint32_t urhs = 0;
+  ASSERT_EQ(counter, urhs);
```

Comment:
Updating Kefu's comments from IRC:

```
<JosCollin> kefu_, badone: So what do you suggest about PR#14705 ? The variable was just to avoid the unnecessary cast.
<kefu_> JosCollin, w.r.t. https://github.com/ceph/ceph/pull/14705, when we "ASSERT_EQ(counter, urhs);", the test is back to single-threaded.
<kefu_> so 1) i don't think it's necessary to check the side-effect of the assignment.
<kefu_> 2) if we must, we'd better putting "ASSERT_EQ(0u, counter);"
```